src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -77,10 +73,90 @@ public record VectorSearchConfig(
+        @Valid
+            @Nullable
+            @JsonInclude(JsonInclude.Include.NON_NULL)
+            @Schema(
+                description = "Optional vectorize configuration to provide embedding service",
+                type = SchemaType.OBJECT,
+                implementation = VectorizeConfig.class)
+            @JsonProperty("service")
+            VectorizeConfig vectorizeConfig) {
+
+      public VectorSearchConfig(Integer dimension, String metric, VectorizeConfig vectorizeConfig) {
         this.dimension = dimension;
         this.metric = metric == null ? "cosine" : metric;
+        this.vectorizeConfig = vectorizeConfig;
+      }
+
+      public record VectorizeConfig(
+          @NotNull
+              @Schema(
+                  description = "Registered Embedding service provider",
+                  type = SchemaType.STRING,
+                  implementation = String.class)
+              @JsonProperty("provider")
+              String provider,
+          @NotNull
+              @Schema(
+                  description = "Registered Embedding service model",
+                  type = SchemaType.STRING,
+                  implementation = String.class)
+              @JsonProperty("model_name")
+              String modelName,
+          @Valid
+              @NotNull
+              @Schema(
+                  description = "Authentication config for chosen embedding service",
+                  type = SchemaType.OBJECT,
+                  implementation = VectorizeServiceAuthentication.class)
+              @JsonProperty("authentication")
+              VectorizeServiceAuthentication vectorizeServiceAuthentication,
+          @Valid
+              @JsonInclude(JsonInclude.Include.NON_NULL)
+              @Nullable
+              @Schema(
+                  description =
+                      "Optional parameters that match the template provided for the provider",
+                  type = SchemaType.OBJECT,
+                  implementation = VectorizeServiceParameter.class)
+              @JsonProperty("parameters")
+              VectorizeServiceParameter vectorizeServiceParameter) {
+        public record VectorizeServiceAuthentication(
+            @NotNull
+                @Schema(
+                    description =
+                        "List of authentications that can be used when sending documents that need vectorization. One or more of \"NONE\", \"HEADER\", \"SHARED_SECRET\"",
+                    type = SchemaType.ARRAY,
+                    implementation = String.class)
+                @JsonProperty("type")
+                List<
+                        @Pattern(
+                            regexp = "(NONE|HEADER|SHARED_SECRET)",
+                            message =
+                                "authentication type can only be one or more of 'NONE', 'HEADER' or 'SHARED_SECRET'")
+                        String>
+                    type,
+            @JsonInclude(JsonInclude.Include.NON_NULL)
```

Comment:
I wouldn't add this until a secrets service is hosted and available

Reply:
Sure, removed it.

Reply:
I will make authentication related stuff as nullable,  since we only have header now.
So basically, you can create a collection like this.

```
{
    "createCollection": {
        "name": "newVectorize1",
        "options": {
            "vector": {
                "metric": "cosine",
                "dimension": 1024,
                "service": {
                    "provider": "nvidia",
                    "model_name": "query"
                    // "authentication": {
                    //     "type": [
                    //         "HEADER"
                    //     ]
                    // },
                    // "parameters": {
                    //     "project_id": "test project"
                    // }
                }
            },
            "indexing":{
                "deny" : ["heh"]
            }
        }
    }
}
```